cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -390,7 +390,9 @@ private Node getNode(Compute compute, Node.Type type, String nodeName) throws IO
       ts = -1L;
     }
     String ip = properties.get("ip.external");
-    if (network.equals(systemNetwork) && !conf.isPreferExternalIP()) {
+    // if the network and the systemNetwork (where data fusion is running are same or they are in network via
+    // network peering and prefer external ip is not set to true then use private ip
+    if ((network.equals(systemNetwork) || conf.isInNetwork()) && !conf.isPreferExternalIP()) {
```

Comment:
This logic was refactored on the release branch to a different place. Can we merge release first and then make this change?

Reply:
Done.